essence-of-live-coding/src/LiveCoding/Cell.lhs

```diff
@@ -412,6 +421,15 @@ Then we can execute the live program in the same way as before.
 
 \begin{comment}
 \begin{code}
+deriving via (WrappedArrow (Cell m)) instance Monad m => Profunctor (Cell m) 
+deriving via (WrappedArrow (Cell m)) instance Monad m => Strong (Cell m)
+deriving via (WrappedArrow (Cell m)) instance Monad m => Data.Profunctor.Choice.Choice (Cell m)
+
+instance Monad m => Traversing (Cell m) where
+  traverse' (Cell state step) = Cell state step' where
+    step' s a = runStateT (traverse (\a -> StateT (`step` a)) a) s
```

Comment:
```suggestion
  traverse' Cell { cellState, cellStep } = Cell
    { cellStep = \s a -> runStateT (traverse (\a -> StateT (`step` a)) a) s
    , ..
    }
```

Reply:
If I apply that change I get the following error:

```
src/LiveCoding/Cell.lhs:431:7: error:
    • Couldn't match expected type ‘Cell m a b’ with actual type ‘s’
      ‘s’ is a rigid type variable bound by
        a pattern with constructor:
          Cell :: forall (m :: * -> *) a b s.
                  Data s =>
                  s -> (s -> a -> m (b, s)) -> Cell m a b,
        in an equation for ‘traverse'’
        at src/LiveCoding/Cell.lhs:429:13-40
    • In the ‘cellState’ field of a record
      In the expression:
        Cell
          {cellStep = \ s a
                        -> runStateT (traverse (\ a -> StateT (`step` a)) a) s,
           ..}
      In an equation for ‘traverse'’:
          traverse' Cell {cellState, cellStep}
            = Cell
                {cellStep = \ s a
                              -> runStateT (traverse (\ a -> StateT (`step` a)) a) s,
                 ..}
    • Relevant bindings include
        cellStep :: s -> a -> m (b, s)
          (bound at src/LiveCoding/Cell.lhs:429:31)
        cellState :: s (bound at src/LiveCoding/Cell.lhs:429:20)
        traverse' :: Cell m a b -> Cell m (f a) (f b)
          (bound at src/LiveCoding/Cell.lhs:429:3)
    |
431 |     , ..
    |       ^^
```

Reply:
I don't see a way to get rid of this error, so I think we can leave it as it is ?